src/foam/blob/RestBlobService.java

```diff
@@ -117,14 +117,12 @@ public Blob find_(X x, Object id) {
 
       if ( connection.getResponseCode() == HttpURLConnection.HTTP_OK ) {
         is = connection.getInputStream();
-        blob = new InputStreamBlob(is);
+        blob = new InputStreamBlob(is, connection.getContentLengthLong());
```

Comment:
We should check getContentLenghtLong() != -1 and return an HTTP 400 perhaps with a reasonable error message if so.

Reply:
Thanks. Done.